tests/models/test_evaluation.py

```diff
@@ -301,7 +304,29 @@ def iris_pandas_df_num_cols_dataset():
     eval_X, eval_y = X[0::3], y[0::3]
     data = pd.DataFrame(eval_X)
     data["y"] = eval_y
-    return EvaluationDataset(data=data, targets="y", name="iris_pandas_df_num_cols_dataset")
+    constructor_args = {"data": data, "targets": "y", "name": "iris_pandas_df_num_cols_dataset"}
+    ds = EvaluationDataset(**constructor_args)
+    ds._constructor_args = constructor_args
+    return ds
+
+
+@pytest.fixture
+def iris_pandas_df_unusual_cols_dataset():
+    X, y = get_iris()
+    eval_X, eval_y = X[0::3], y[0::3]
+    data = pd.DataFrame(
+        {
+            "f1": eval_X[:, 0],
+            2: eval_X[:, 1],
+            "f3longnamelongnamelongname": eval_X[:, 2],
+            123456789012345678901234567890: eval_X[:, 3],
+            "y": eval_y,
+        }
+    )
```

Comment:
Can we use a numpy array?

Reply:
Using a numpy array would test the use of unlabelled columns with explainability, but would not test the truncation feature on long string names

Reply:
Ok, can we create two datasets, one for testing unlabelled columns (numpy array) with explainability, and one for testing truncation feature on long string names?

Reply:
Yes